pkg/controller/controller_test.go

```diff
@@ -79,9 +81,10 @@ var _ = Describe("Controller", func() {
 
 			It(test.descr, func() {
 				Expect(controller.ProcessNextItem()).To(BeTrue())
-				pod, err := fakeClient.CoreV1().Pods("").Get(test.expectPodName, metav1.GetOptions{})
-				Expect(err).NotTo(HaveOccurred())
-				Expect(pod.Name).To(Equal(test.expectPodName))
+				pod, err := getTestPod(fakeClient, test.expectPodName)
+				Expect(err).To(BeNil(), fmt.Sprintf("Expected nil err, got: %v", err))
+				Expect(pod).NotTo(BeNil(), fmt.Sprintf("Expected Pod %q was not found.", test.expectPodName))
+				Expect(pod.GenerateName).To(ContainSubstring(test.expectPodName))
```

Comment:
Why are the `BeforeEach` and `AfterEach` (L68, L78) inside the `for` loop? Can't these be defined under the `Context`?